util/entitlements/security/security_linux.go

```diff
@@ -160,3 +127,68 @@ func getFreeLoopID() (int, error) {
 	}
 	return 0, errors.Errorf("error getting free loop device: %v", uerr)
 }
+
+func getAllCaps() ([]string, error) {
+	// allCaps provides a list of capabilities to be granted, some of the capabilities might not be available
+	// if that's the case, capabilities are skipped and warning is printed
+	allCaps := []string{
+		"CAP_FSETID",
+		"CAP_KILL",
+		"CAP_FOWNER",
+		"CAP_MKNOD",
+		"CAP_CHOWN",
+		"CAP_DAC_OVERRIDE",
+		"CAP_NET_RAW",
+		"CAP_SETGID",
+		"CAP_SETUID",
+		"CAP_SETPCAP",
+		"CAP_SETFCAP",
+		"CAP_NET_BIND_SERVICE",
+		"CAP_SYS_CHROOT",
+		"CAP_AUDIT_WRITE",
+		"CAP_MAC_ADMIN",
+		"CAP_MAC_OVERRIDE",
+		"CAP_DAC_READ_SEARCH",
+		"CAP_SYS_PTRACE",
+		"CAP_SYS_MODULE",
+		"CAP_SYSLOG",
+		"CAP_SYS_RAWIO",
+		"CAP_SYS_ADMIN",
+		"CAP_LINUX_IMMUTABLE",
+		"CAP_SYS_BOOT",
+		"CAP_SYS_NICE",
+		"CAP_SYS_PACCT",
+		"CAP_SYS_TTY_CONFIG",
+		"CAP_SYS_TIME",
+		"CAP_WAKE_ALARM",
+		"CAP_AUDIT_READ",
+		"CAP_AUDIT_CONTROL",
+		"CAP_SYS_RESOURCE",
+		"CAP_BLOCK_SUSPEND",
+		"CAP_IPC_LOCK",
+		"CAP_IPC_OWNER",
+		"CAP_LEASE",
+		"CAP_NET_ADMIN",
+		"CAP_NET_BROADCAST",
+	}
+
+	availableCaps, err := cap.Current()
+	if err != nil {
+		return nil, fmt.Errorf("error getting available capabilities: %s", err)
+	}
+
+	// sort availableCaps to perform binary search on them
+	sort.Strings(availableCaps)
+
+	addCaps := make([]string, 0, len(allCaps))
+
+	for _, cap := range allCaps {
```

Comment:
I think we should take all `availableCaps` and warn if a known cap is missing from the `Current` list. The known caps list probably makes more sense as a map for this.

Reply:
sorry, I think I got confused on what we're going to achieve.

* before my change we had a hardcoded list of caps
* with my change we filter out caps from the list which are not available on the worker (and warns about any capability filtered out)

Do we want to do what:

* expand the list to `cap.Known()` ?
* use the `cap.Current` as the list and warn if any of hardcoded are missing there?

My goal was to make sure there's no change in list of capabilities if there are no restrictions

Reply:
this might be tricky for the tests we have, as "all the caps that system supports" depends on the system where we run the tests. we can still try to pull that through in the tests, but we'd have to do same calculation as we have in the code to see what we should come up with